scheduler/planner.go

```diff
@@ -281,13 +260,74 @@ func (unit *Unit) RankValue() int64 {
 	// Increase the value for the number of dependencies, so that
 	// tasks (and units) which block other tasks run before tasks
 	// that don't block other tasks.
-	unit.cachedValue += priority * (numDeps / length)
+	value += priority * (u.NumDeps / length)
 
 	// Increase the value for tasks with longer runtimes, given
 	// that most of our workloads have different runtimes, and we
 	// don't want to have longer makespans if longer running tasks
 	// have to execute after shorter running tasks.
-	unit.cachedValue += priority * unit.distro.GetExpectedRuntimeFactor() * int64(math.Floor(expectedRuntime.Minutes()/float64(length)))
+	value += priority * u.Settings.GetExpectedRuntimeFactor() * int64(math.Floor(u.ExpectedRuntime.Minutes()/float64(length)))
+
+	return value
+}
+
+func (unit *Unit) info() unitInfo {
+	info := unitInfo{
+		Settings: unit.distro.PlannerSettings,
+	}
+
+	for _, t := range unit.tasks {
+		if t.Requester == evergreen.MergeTestRequester {
+			info.InCommitQueue = true
+		} else if evergreen.IsPatchRequester(t.Requester) {
+			info.InPatch = true
+		}
+
+		if t.TaskGroup == "" {
+			info.AnyNonGroupTasks = true
+		}
+		if t.GenerateTask {
+			info.GenerateTask = true
+		}
+		if t.ActivatedBy == evergreen.StepbackTaskActivator {
+			info.StepbackTask = true
+		}
+
+		if !t.ActivatedTime.IsZero() {
+			info.TimeInQueue += time.Since(t.ActivatedTime)
+		} else if !t.IngestTime.IsZero() {
+			info.TimeInQueue += time.Since(t.IngestTime)
+		}
+
+		info.TotalPriority += t.Priority
+		info.ExpectedRuntime += t.FetchExpectedDuration().Average
+		info.NumDeps += int64(t.NumDependents)
+		info.TaskIDs = append(info.TaskIDs, t.Id)
+	}
+
+	return info
+}
+
+// RankValue returns a point value for the tasks in the unit that can
+// be used to compare units with each other.
+//
+// Generally, higher point values are given to larger units and for
+// units that have been in the queue for longer, with longer expected
+// runtimes. The tasks' priority acts as a multiplying factor.
+func (unit *Unit) RankValue() int64 {
+	if unit.cachedValue > 0 {
+		return unit.cachedValue
+	}
+
+	info := unit.info()
+	unit.cachedValue = info.value()
+
+	grip.Info(message.Fields{
+		"message":   "prioritizated distro unit",
```

Comment:
prioritized I'm guessing, unless we're coining an new word 🤪

Reply:
😅 
fixed